source/features/sync-pr-commit-title.tsx

```diff
@@ -77,31 +77,24 @@ async function updateCommitTitle(): Promise<void> {
 	updateUI();
 }
 
-function disableSubmission(): void {
-	deinit();
-	getUI().remove();
-}
-
-const subscriptions: delegate.Subscription[] = [];
-
-function init(): Deinit {
-	subscriptions.push(
-		onPrCommitMessageRestore(updateUI),
-		onPrMergePanelOpen(updateCommitTitle),
-		delegate(document, '#merge_title_field', 'input', updateUI),
-		delegate(document, 'form.js-merge-pull-request', 'submit', updatePRTitle),
-		delegate(document, '.rgh-sync-pr-commit-title', 'click', disableSubmission),
-	);
-
-	return deinit;
-}
-
-function deinit(): void {
-	for (const subscription of subscriptions) {
-		subscription.destroy();
-	}
-
-	subscriptions.length = 0;
+function init(deinitSignal: AbortSignal): void {
+	const internalController = new AbortController();
+	const {signal} = internalController;
+	deinitSignal.addEventListener('abort', () => {
+		if (!signal.aborted) {
+			internalController.abort();
+		}
+	});
+
+	onPrCommitMessageRestore(updateUI, signal);
+	onPrMergePanelOpen(updateCommitTitle, signal);
+
+	delegate(document, '#merge_title_field', 'input', updateUI, {signal});
+	delegate(document, 'form.js-merge-pull-request', 'submit', updatePRTitle, {signal});
+	delegate(document, '.rgh-sync-pr-commit-title', 'click', () => {
+		internalController.abort();
+		getUI().remove();
+	}, {signal});
```

Comment:
This code is kind of awkward, there's probably a better solution :thinking:

Reply:
If Safari wasn't a party-pooper (it doesn't bind `.abort`) we could use

```js
 	delegate(document, '.rgh-sync-pr-commit-title', 'click', internalController.abort, {signal});
	onAbort(internalController, () => {
		getUI().remove()
	})
```

Either way not a great improvement